pkg/cluster/internal/providers/docker/provision.go

```diff
@@ -199,6 +199,22 @@ func commonArgs(cluster string, cfg *config.Cluster, networkName string, nodeNam
 	if mountFuse() {
 		args = append(args, "--device", "/dev/fuse")
 	}
+
+	if cfg.Networking.DNSSearch != nil {
+		if len(*cfg.Networking.DNSSearch) == 0 {
+			return nil, errors.New("docker provider requires a non-empty dnsSearch")
+		}
+		for _, s := range *cfg.Networking.DNSSearch {
+			args = append(args, "--dns-search", s)
+		}
+		args = append(args, "--dns", "8.8.8.8")
+		args = append(args, "--dns", "8.8.4.4")
+		if config.ClusterHasIPv6(cfg) {
+			args = append(args, "--dns", "2001:4860:4860::8888")
+			args = append(args, "--dns", "2001:4860:4860::8844")
+		}
```

Comment:
I have 100% no clue why this works. Without it, things DNS gets blackholed. With it, things work perfectly and these don't actually end up in `/etc/resolv.conf` at all...

Reply:
it seems an internal thing at first sight .. we should not set `--dns`, should we?

Reply:
I don't think we should set it, but somehow if we don't it doesn't work. I don't 100% understand why since even when we set it, it doesn't actually end up in.

I just found its something with `kind` network I guess, so will explore that more

```
$ docker run --dns-search=com --dns 8.8.8.8 howardjohn/nettools cat /etc/resolv.conf
search com
nameserver 8.8.8.8
options edns0 trust-ad
$ docker run --dns-search=com --dns 8.8.8.8 --dns 8.8.4.4 --network kind howardjohn/nettools cat /etc/resolv.conf
search com
nameserver 127.0.0.11
options edns0 trust-ad ndots:0
```

Reply:
I think there is some magic when you use non-default network that it configures the internal dns server. And when you DON'T set any dns configs it automatically replaces 127.0.0.1 from host with the google ones, but if you set and `--dns-*` flags that magic is gone... 

Low confidence

Reply:
Updated approach and PR description with why